pkg/server/event_handler_test.go

```diff
@@ -393,17 +394,32 @@ isOK: true`),
 			successMessenger: &testMessenger{
 				gotPmapEvent: &v1alpha1.PmapEvent{},
 			},
+			wantErrSubstr: "always fail",
 			wantPmapEvent: &v1alpha1.PmapEvent{},
 		},
 		{
-			name: "failed_process_and_send",
+			name: "failed_process_with_processErr",
+			notification: &pubsub.Message{
+				Attributes: map[string]string{"bucketId": "foo", "objectId": "pmap-test/gh-prefix/dir1/dir2/bar"},
+				Data:       testGCSMetadataBytes(),
+			},
+			gcsObjectBytes: []byte(`foo: bar
+isOK: true`),
+			processors: []Processor[*structpb.Struct]{&failProcessor{fmt.Errorf("processErr")}},
```

Comment:
Don't wrap the error in the `failProcessor`. Instead of inject the wrapped error here.
```suggestion
			processors: []Processor[*structpb.Struct]{&failProcessor{pmaperrors.New("user facing error)},
```